tools/autograd/templates/Functions.cpp

```diff
@@ -430,6 +430,19 @@ Tensor solve_backward_A(const Tensor & grad, const Tensor & self, const Tensor &
   return -at::matmul(grad_self, solution.transpose(-2, -1));
 }
 
+Tensor logcumsumexp_backward(Tensor grad, const Tensor & self, Tensor result, int64_t dim) {
+  grad = grad*(-result).exp();
+  if (grad.dim() == 0 || grad.numel() == 0) {
```

Comment:
this condition should precede the previous line. it should be the first condition to be checked before we begin any computations